Source/WebCore/page/Navigation.cpp

```diff
@@ -290,7 +306,7 @@ Navigation::Result Navigation::performTraversal(const String& key, Navigation::O
     // FIXME: This is just a stub that loads a URL for now.
     auto entry = findEntryByKey(key);
     ASSERT(entry);
-    frame()->loader().load(FrameLoadRequest(*frame(), URL(entry.value()->url())));
+    frame()->loader().loadItem(entry.value()->associatedHistoryItem(), &currentEntry()->associatedHistoryItem(), loadType, ShouldTreatAsContinuingLoad::No);
```

Comment:
Should we be using currentEntry and entry here?  Can they be different.

Reply:
First value is `target`, second value is `from` (the current).

The case of being the same is handled above, so they are always different.